src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/nodes/interrupts/CheckForInterruptsState.java

```diff
@@ -74,39 +78,70 @@ public void shutdown() {
         }
     }
 
-    public void setInterruptPending() {
-        interruptPending = true;
-        shouldTrigger = isActive;
+    @TruffleBoundary
+    private void createOrUpdateInterruptChecks() {
+        if (executor != null) {
+            if (interruptChecks != null && !interruptChecks.isCancelled()) {
+                interruptChecks.cancel(false);
+            }
+            interruptChecks =
+                    executor.scheduleWithFixedDelay(() -> {
+                        if (!shouldTrigger) setShouldTriggerIfNeeded();
```

Comment:
I think we should move `!shouldTrigger` into `setShouldTriggerIfNeeded` and avoid the additional lambda. Just pass `setShouldTriggerIfNeeded` to `executor.scheduleWithFixedDelay()`.

Reply:
Done!